secure/storage/src/storage.rs

```diff
@@ -31,4 +38,36 @@ pub trait Storage: Send + Sync {
     fn get(&self, key: &str) -> Result<Value, Error>;
     /// Sets a value in storage and fails if invalid permissions or it does not exist
     fn set(&mut self, key: &str, value: Value) -> Result<(), Error>;
+
+    /// Securely generates a new named Ed25519 key pair and returns the corresponding public key.
+    ///
+    /// The new key pair is named according to the 'key_pair_name' and is held in secure storage
+    /// under the given 'policy'. To access or use the key pair (e.g., sign or encrypt data),
+    /// subsequent API calls must refer to the key pair by name. As this API call may fail
+    /// (e.g., if a key pair with the given name already exists), an error may also be returned.
+    ///
+    /// This default implementation is useful for storage engines that only implement the simple
+    /// Storage interface. More complex engines (e.g., those that support native crypto key
+    /// generation), should override this implementation.
+    fn generate_new_ed25519_key_pair(
+        &mut self,
+        key_pair_name: &str,
+        policy: &Policy,
+    ) -> Result<Ed25519PublicKey, Error> {
+        let mut seed_rng = OsRng::new().map_err(|e| Error::EntropyError(e.to_string()))?;
+        let mut rng = rand::rngs::StdRng::from_seed(seed_rng.gen());
+        let private_key = Ed25519PrivateKey::generate_for_testing(&mut rng);
+        let public_key = private_key.public_key();
+        self.create(key_pair_name, Value::Ed25519PrivateKey(private_key), policy)
+            .map(|_| public_key)
+    }
+    /// Returns the corresponding public key for a given Ed25519 key pair, as identified by the
```

Comment:
nit new line

Reply:
Done, thanks. (I was trying to be consistent as all K/V related operations don't have new lines and instead only have a newline that separates K/V operations and crypto operations, but seeing as though that didn't come across, I'll just fix this!)